core/templates/pages/diagnostic-test-player-page/diagnostic-test-player.component.ts

```diff
@@ -176,4 +185,18 @@ export class DiagnosticTestPlayerComponent implements OnInit {
       }
     );
   }
+  getRecommendationAcceptanceEvent(topicName: string): void {
+    if (this.classroomData) {
+      const topicSummary = this.recommendedTopicSummaries.find(
+        summary => summary.getName() === topicName
+      );
+      if (topicSummary) {
+        const topicId = topicSummary.getId();
+        this.siteAnalyticsService.registerRecommendationAcceptedEvent(
```

Comment:
@himkar-cmd Could you have "diagnostic test" in the name somewhere -- e.g. registerDiagnosticTestRecommendationAcceptedEvent? There are other recommendations throughout the platform (e.g. which lesson to do next) so the existing name is too generic.